### PR TITLE
Show supervisor addon configuration error

### LINKED ISSUE #2822: Unable to edit Add-On Config in fresh VM install
**Hardware Environment**

ESXi 7.0.1

**Home Assistant OS release:**
https://github.com/home-assistant/operating-system/releases/download/5.13/hassos_ova-5.13.ova

**Description of problem:**

1. Create a new VM using default settings in ESXi 7.0.1 using https://github.com/home-assistant/operating-system/releases/download/5.13/hassos_ova-5.13.ova, with all default settings
2. When VM starts complete initial setup and install an Add On such as Duck DNS
3. Edit Add On Configuration in yaml mode using the supervisor frontend web ui on any web browser of your choosing
4. When configuration is saved, it reverts to default settings

I've  been able to re-create this multiple times with mutable Add Ons with multiple fresh VMs 
but some Add Ons work as expected such as Terminal & SSH

### PR DESCRIPTION
**Pull request by ludeeus**
## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes home-assistant/supervisor#2822
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: hassio/src/addon-view/config/hassio-addon-config.ts [hunk 1]
```diff
@@ -266,33 +266,43 @@ class HassioAddonConfig extends LitElement {
...
-      const eventdata = {
-        success: true,
-        response: undefined,
-        path: "options",
-      };
-      fireEvent(this, "hass-api-called", eventdata);
       if (this.addon?.state === "started") {
         await suggestAddonRestart(this, this.hass, this.supervisor, this.addon);
       }
     } catch (err) {
```
Comment (bramkragten) [inline hassio/src/addon-view/config/hassio-addon-config.ts:296]: When does it throw, and when not on error? And why does it not always throw?

**Review by bramkragten — APPROVED** [review]

File: hassio/src/addon-view/config/hassio-addon-config.ts (hunk 1 above)
Comment (ludeeus) [reply hassio/src/addon-view/config/hassio-addon-config.ts:296]: It throws on request failure only

Comment (bramkragten) [reply hassio/src/addon-view/config/hassio-addon-config.ts:296]: Would it make sense to have `setHassioAddonOption` throw when `response.result === "error"`?

Comment (ludeeus) [reply hassio/src/addon-view/config/hassio-addon-config.ts:296]: Sure

File: src/data/hassio/addon.ts [hunk 2]
```diff
@@ -186,16 +190,20 @@ export const setHassioAddonOption = async (
...
-    return;
+
+    if (response.result === "error") {
+      throw Error(extractApiErrorMessage(response));
+    }
+    return response;
   }
 
-  await hass.callApi<HassioResponse<void>>(
+  return await hass.callApi<HassioResponse<any>>(
```
Comment (bramkragten) [inline src/data/hassio/addon.ts:206]:
```suggestion
  return hass.callApi<HassioResponse<any>>(
```

File: src/data/hassio/addon.ts [hunk 3]
```diff
@@ -186,16 +190,20 @@ export const setHassioAddonOption = async (
...
-    await hass.callWS({
+    const response = await hass.callWS<HassioResponse<any>>({
       type: "supervisor/api",
       endpoint: `/addons/${slug}/options`,
       method: "post",
       data,
     });
-    return;
+
+    if (response.result === "error") {
```
Comment (bramkragten) [inline src/data/hassio/addon.ts:200]: So only the new API returns "error" as result?

Comment (ludeeus) [reply src/data/hassio/addon.ts:200]: yes, the old just fails

**Review by bramkragten — APPROVED** [review]

**Pull request merged**